### PR TITLE
Fix DASH ACL implementation and add testcases

### PR DESCRIPTION
**Pull request by Pterosaur**
BUG 1: An ACL rule cannot be inserted to the corresponding ACL table.
In the previous implementation of DASH ACL SAI, the ACL rule will always be added to a single ACL table, Because the ID of single ACL table was hardcode into the function `sai_create_dash_acl_rule`.
In this PR, I combined all dash ACL tables with same table name and added one ACL rule to all ACL tables.

BUG 2: The L4 port cannot be fetched in ACL table.
In the previous implementation, the ACL table use tcp port as a key so that the port of UDP packet cannot be fetched correct.
In this PR, I add two new fields in metadata to collect the L4 port as the key of ACL table.

More testcases for ACL will be added in later PR.

Fix issue: https://github.com/Azure/DASH/issues/232

### CODE REVIEW

**Comment by aputriax** [comment]
Hi! Please consider putting your test plan to the standard location: test/docs/testplans/acl.md
See my open PR #127.
Thank you!

**Comment by Pterosaur** [comment]
> Hi! Please consider putting your test plan to the standard location: test/docs/testplans/acl.md See my open PR #127. Thank you!

Hi @aputriax , thanks for your reminding, I will move it.

**Comment by chrispsommers** [comment]
Hi @Pterosaur For some reason the CI pipeline seems stuck in "Run PTF Tests." This is true in your fork as well as the PR. See here: https://github.com/Azure/DASH/actions/runs/3480389093/jobs/5820119909. It's been stuck in this step w/o any output for almost 2 hours. Have you successfully executed `make run-all-tests` in your dev environment?

Also, note that https://github.com/Azure/DASH/pull/281 is in review but once it is merged, you will need to change your PTF test base classes to `SaiHelperSimplified`. It should not have any functional impact, but it will bring it in alignment with other tests. Thanks for working on this!

File: dash-pipeline/tests/saithrift/ptf/acl/test_saithrift_acl.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+            print("Verifying packet...\n", pkt_exp.__repr__())
+            verify_packet(self.saithrift, pkt_exp, 0)
+        else:
+            print("Verifying drop...")
+            verify_no_other_packets(self.saithrift)
+        print("\n")
+        print("Acl test {} OK".format(self.meta))
+
+
+class TestSaiThrift_dash_acl(ThriftInterfaceDataPlane):
```
Comment (chrispsommers) [inline dash-pipeline/tests/saithrift/ptf/acl/test_saithrift_acl.py:66]: Change base class to `SaiHelperSimplified` once https://github.com/Azure/DASH/pull/281 is merged.

File: dash-pipeline/tests/saithrift/ptf/acl/test_saithrift_acl.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+    def clearup_objs(self):
+        if self.cleaned_up:
+            return
+        for remove_func, obj in self.teardown_stack:
+            assert (remove_func(self.client, obj) == SAI_STATUS_SUCCESS)
+        self.cleaned_up = True
+
+    def tearDown(self):
+        self.clearup_objs()
```
[the remark this answers is not in the document]
Comment (Pterosaur) [reply dash-pipeline/tests/saithrift/ptf/acl/test_saithrift_acl.py:230]: Thanks for your very useful suggestion.

File: dash-pipeline/tests/saithrift/ptf/acl/test_saithrift_acl.py (hunk 1 above)
Comment (Pterosaur) [reply dash-pipeline/tests/saithrift/ptf/acl/test_saithrift_acl.py:66]: Thanks for your information.

**Comment by chrispsommers** [comment]
@Pterosaur if this is to fix https://github.com/Azure/DASH/issues/232 please indicate in description, it will create a cross-reference. Thanks.

**Comment by Pterosaur** [comment]
> @Pterosaur if this is to fix #232 please indicate in description, it will create a cross-reference. Thanks.

@chrispsommers No problem.

**Review by lguohan — APPROVED** [review]

File: dash-pipeline/tests/saithrift/ptf/acl/test_saithrift_acl.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    def create_entry(self, create_func, remove_func, entry, *args, **kwargs):
+        status = create_func(self.client, entry, *args, **kwargs)
+        assert (status == SAI_STATUS_SUCCESS)
+        self.teardown_stack.append((remove_func, entry))
+        return status
+
+    def create_obj(self, create_func, remove_func, *args, **kwargs):
+        obj = create_func(self.client, *args, **kwargs)
+        assert (obj != SAI_NULL_OBJECT_ID)
+        self.teardown_stack.append((remove_func, obj))
```
Comment (aputriax) [inline dash-pipeline/tests/saithrift/ptf/acl/test_saithrift_acl.py:78]: This approach is already used here: https://github.com/Azure/DASH/blob/f64277566c62145f2a20f18fd36b4f5a1c1ba8d6/test/test-cases/functional/saidashvnet.py#L59

We need to find a common way how to do this.

File: dash-pipeline/tests/saithrift/ptf/acl/test_saithrift_acl.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+        return obj
+
+    def setUpSwitch(self):
+        self.switch_id = 5
```
Comment (aputriax) [inline dash-pipeline/tests/saithrift/ptf/acl/test_saithrift_acl.py:82]: This should be already defined once you have switched to `SaiHelperSimplified` as Chris wrote.

File: dash-pipeline/tests/saithrift/ptf/acl/test_saithrift_acl.py [hunk 5]
```diff
@@ -0,0 +1,231 @@
+import copy
```
Comment (aputriax) [inline dash-pipeline/tests/saithrift/ptf/acl/test_saithrift_acl.py:1]: We might need to move this file in a new location once PR #277 is merged.

**Comment by Pterosaur** [comment]
Hi @aputriax , thanks for your comments. I will rebase my PR to the latest once these pre-PRs were merged.

**Review by marian-pritsak — APPROVED** [review]
The loop over stages in `sai_status_t sai_create_{{ table.name }}(` results in generating the same code for parsing attributes for every stage. I'm not sure that's the best way to do it, but, on the other hand, it's still a correct behavior.

Other than that, looks good to me.

**Pull request merged**